sklearn/cluster/tests/test_k_means.py

```diff
@@ -824,3 +824,28 @@ def test_KMeans_init_centers():
         km = KMeans(init=init_centers_test, n_clusters=3, n_init=1)
         km.fit(X_test)
         assert_equal(False, np.may_share_memory(km.cluster_centers_, init_centers))
+
+
+def test_sparseX_KMeans_init_centers():
+    from sklearn.datasets import load_iris
+
+    iris = load_iris()
+    X = iris.data
+
+    # Get a local optimum
+    centers = KMeans(n_clusters=3).fit(X).cluster_centers_
+
+    # Fit starting from a local optimum shouldn't change the solution
+    np.testing.assert_allclose(
+        centers,
+        KMeans(n_clusters=3, init=centers, n_init=1).fit(X).cluster_centers_
+    )
+
+    # The same should be true when X is sparse
+    X_sparse = sp.csr_matrix(X)
+    np.testing.assert_allclose(
+        centers,
+        KMeans(n_clusters=3,
+               init=centers,
```

Comment:
maybe add a test that an error is raised if init has the wrong shape (say 4 custers)